### ISSUE TITLE
Remove usage of FilePath.statinfo in twisted

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4711]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)| @cyli reported|
|-|-|
|Trac ID|trac#4711|
|Type|defect|
|Created|2010-10-25 14:58:10Z|
|Branch|https://github.com/twisted/twisted/tree/remove-statinfo-in-ftp-4711|

FilePath.statinfo should be deprecated (see [#4450](https://github.com/twisted/twisted/issues/4450)) and thus it should no longer be used in Twisted.

It is still used in twisted.protocols.ftp, and twisted.web2.static

<details><summary>Searchable metadata</summary>

```
trac-id__4711 4711
type__defect defect
reporter__cyli cyli
priority__normal normal
milestone__ 
branch__branches_remove_statinfo_in_ftp_4711 branches/remove-statinfo-in-ftp-4711
branch_author__cyli cyli
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1288018690000000 1288018690000000
changetime__1397864699000000 1397864699000000
version__None None
owner__cyli cyli

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_1"></a>|@cyli commented|
|-|-|

(In [30173]) Branching to 'remove-usages-of-filepath-statinfo-4711'

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_2"></a>|@cyli commented|
|-|-|

Usage of FilePath.statinfo cannot be removed yet because FilePath does not provide accessors for some of the fields.  So this ticket is on hold until [#4712](https://github.com/twisted/twisted/issues/4712) is resolved.

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_3"></a>|@cyli commented|
|-|-|

(In [30516]) Branching to 'begin-convert-usages-of-filepath-statinfo-4711'

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_4"></a>|@cyli commented|
|-|-|

(In [30517]) Added a test to ensure that the new accessors for statinfo will work in place of statinfo in twisted/web2/static.py

refs [#4711](https://github.com/twisted/twisted/issues/4711)

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_5"></a>|@cyli commented|
|-|-|

(In [30537]) Changed FTP to (mostly) use the new accessor methods in t.p.filepath. However, an accessor for permissions is missing.

refs [#4711](https://github.com/twisted/twisted/issues/4711)

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_6"></a>|@cyli commented|
|-|-|

Filed ticket [#4813](https://github.com/twisted/twisted/issues/4813) for filepath to provide the permissions

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)|@cyli set owner to @cyli<br>@cyli set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_9"></a>|@itamarst commented|
|-|-|

Can this ticket be finished now that [#4813](https://github.com/twisted/twisted/issues/4813) is closed?

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_10"></a>|@cyli commented|
|-|-|

(In [42358]) Branching to remove-statinfo-in-ftp-4711.

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_11"></a>|@cyli set status to `new`|
|-|-|

Sorry for not getting back to this in a while.  So, getUserID(), getGroupID(), and getNumberOfHardLinks() are not implemented for windows in filepath, so this changes the behavior of FTP a bit, because it just returns hardlinks: 0, user: "0", group: "0" in windows, since that's probably the previous behavior (stat in windows seems to set those values to 0 anyway: http://stackoverflow.com/questions/5275731/os-stat-on-windows)

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_12"></a>|@adiroiban commented|
|-|-|

Looks good to merge. Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_13"></a>|@glyph set owner to @cyli|
|-|-|

I concur with Adi's analysis.  Thanks for eliminating this unpleasant implementation detail from ftp.

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_14"></a>|@cyli set status to `closed`|
|-|-|

(In [42473]) Merge remove-statinfo-in-ftp-4711

Author: cyli
Reviewer: adiroiban, glyph
Fixes: [#4711](https://github.com/twisted/twisted/issues/4711)

Removes the final usage of twisted.python.filepath.FilePath.statinfo from the Twisted codebase
